### DISCUSSION TITLE
Using TabulatedFluidProperties with FlowChannel1Phase

### DISCUSSION THREAD

**Discussion by jcstonehillAMA** [idaholab/moose#21552] [Q&A General] [answered]
Hello all,

Hopefully this is a simple fix: I'm attempting to build a simulation that uses many components from the Thermal Hydraulics module include the FlowChannel1Phase. In addition, I have some flow properties that I have saved in a CSV file (completely separate from MOOSE) that I would like to use instead of an IdealGasFluidProperties object.

As soon as I set everything up and run, I get this error: 

```
*** ERROR ***
The following error occurred in the object "tabulated_fluid_data", of type "TabulatedFluidProperties".

tabulated_fluid_data: virtual libMesh::Real SinglePhaseFluidProperties::e_from_p_rho(libMesh::Real, libMesh::Real) const not implemented.
```

Here is the portion of the input file where I set up my fluid properties:

```
[Modules]
  [./FluidProperties]
    [./original]
                       type = IdealGasFluidProperties
			gamma = 1.3845
			k = 0.19158
			molar_mass = 0.00201588
			mu = 8.9154e-5
    [../]
    [./tabulated_fluid_data]
			type = TabulatedFluidProperties
			fluid_property_file = fluid_properties.csv
			fp = original
			pressure_min = 1
			pressure_max = 100000000
			temperature_min = 1
			temperature_max = 6000
     []
   []
[]
```

And here's my flow channel component:
```
[fch]
		type = FlowChannel1Phase
		position = '0 0 0'
		orientation = '0 0 1'
		fp = tabulated_fluid_data
		n_elems = 6
		length = ${L}
		initial_T = 1000
		initial_p = 2.3e6
		initial_vel = 0
		closures = simple_closures
		A = 0.000005309
		f = 0
[]
```

Is it possible to load in this custom fluid properties data? and if so, how?

Many thanks ahead of time!

EDIT: Sorry for the terribly formatted code, my IDE did not like that copy and paste for some reason.

**Comment by GiudGiud** [comment, marked as answer]
Hello

It is possible in a pull request, but not in the current version of MOOSE. @spaubt2  and I are actively working on making this work. 
e_from_p_rho is currently not implemented for this fluid property in MOOSE, but we have it working well in this pull request
https://github.com/idaholab/moose/pull/21301

@spaubt2 can you please upload the latest working version to the PR so @jcstonehillAMA may try it

Guillaume
